File: packages/selfie-segmentation/ios/Plugin/Classes/ProcessImageResult.swift
import Foundation
import Capacitor

@objc class ProcessImageResult: NSObject {
    let image: UIImage

    init(image: UIImage) {
        self.image = image
    }

    func toJSObject() throws -> JSObject {
        var result = JSObject()

        if let data = image.pngData() {
            let uniqueFileNameWithExtension = UUID().uuidString + ".png"
            var directory = URL(fileURLWithPath: NSTemporaryDirectory())
            if let cachesDirectory = FileManager.default.urls(for: .cachesDirectory, in: .userDomainMask).first {
                directory = cachesDirectory
            }
            let url = directory.appendingPathComponent(uniqueFileNameWithExtension)
            try data.write(to: url)

            result["path"] = url.absoluteString
            result["width"] = Int(image.size.width)
            result["height"] = Int(image.size.height)
        }

        return result
    }
}

File: packages/subject-segmentation/android/src/main/java/io/capawesome/capacitorjs/plugins/mlkit/subjectsegmentation/classes/ProcessImageResult.java
package io.capawesome.capacitorjs.plugins.mlkit.subjectsegmentation.classes;

import com.getcapacitor.JSObject;

public class ProcessImageResult {

    private final String imagePath;

    private final int width;
    private final int height;

    public ProcessImageResult(String imagePath, int width, int height) {
        this.imagePath = imagePath;

        this.width = width;
        this.height = height;
    }

    public JSObject toJSObject() {
        JSObject result = new JSObject();

        result.put("path", imagePath);

        result.put("width", width);
        result.put("height", height);

        return result;
    }
}
